src/spring-cloud/azext_spring_cloud/tanzu.py

```diff
@@ -205,3 +226,30 @@ def _assert_deployment_exist_and_retrieve_name(cmd, client, resource_group, serv
         raise CLIError('Deployment not found, create one by running "az spring-cloud tanzu app " \
             "update -g {} -s {} -n {}"'.format(resource_group, service, name))
     return deployment.name
+
+
+def _set_pattern_for_deployment(patterns):
+    return {
+        TANZU_CONFIGURATION_SERVICE_NAME: models.AddonProfile(
+            properties = {
+                TANZU_CONFIGURATION_SERVICE_PROPERTY_PATTERN: patterns
+            }
+        )
+    }
+
+
+def _tcs_bind_or_unbind_app(cmd, client, resource_group, service, app_name, enabled):
+    try:
+        app = _app_get(cmd, client, resource_group, service, app_name)
+        previous = app.properties.addon_config[TANZU_CONFIGURATION_SERVICE_NAME].enabled
+        app.properties.addon_config[TANZU_CONFIGURATION_SERVICE_NAME] = models.AddonProfile(
+            enabled=enabled
+        )
+        if (enabled != previous):
+            return client.apps.create_or_update(resource_group, service, app_name, app.properties)
+        elif (enabled):
+            logger.warning('app {} has been binded'.format(app_name))
+        else:
+            logger.warning('app {} has been unbinded'.format(app_name))
+    except CloudError:

```

Comment:
`_app_get` will throw CLIError App not found. no need to try catch here.

Reply:
Yes, but the error will be "App not found", without app name.